lib/savon/response.rb

```diff
@@ -54,7 +54,7 @@ def to_array(*path)
       result.kind_of?(Array) ? result.compact : [result].compact
     end
 
-    def hash
+    def response_hash
       @hash ||= nori.parse(xml)
```

Comment:
Minor: often, when memoizing values in methods like this, we call the instance variable the same name as the method.

```suggestion
      @response_hash ||= nori.parse(xml)
```